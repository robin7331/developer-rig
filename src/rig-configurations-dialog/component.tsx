import * as React from 'react';
import * as closeButton from '../img/close_icon.png';
import './component.sass';

interface RigConfigurationsDialogProps {
  config: object;
  closeConfigurationsHandler: () => void;
  refreshConfigurationsHandler: () => void;
}

export class RigConfigurationsDialog extends React.Component<RigConfigurationsDialogProps> {
  public render() {
    return (
      <div className="rig-configurations-view">
        <div className="rig-configurations-view__background"/>
        <div className="rig-configurations-view__dialog">
          <div className="dialog__top-bar-container">
            <div className="top-bar-container__title"> Dev Rig Configurations </div>
            <div className="top-bar-container__escape" onClick={this.props.closeConfigurationsHandler}><img alt="Close" src={closeButton}/></div>
          </div>
          <hr className="dialog__divider"/>
          <div className="rig-configurations-view__content"> {JSON.stringify(this.props.config, null, 2)} </div>
          <hr className="dialog__divider"/>
          <div className="dialog_bottom-bar">
            <div className="bottom-bar__refresh" onClick={this.props.refreshConfigurationsHandler}> Refresh </div>
            <div className="bottom-bar__cancel" onClick={this.props.closeConfigurationsHandler}> Cancel </div>
          </div>
        </div>
      </div>
    );
  }
}

