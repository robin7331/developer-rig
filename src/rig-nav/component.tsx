import * as React from 'react';
import * as classNames from 'classnames';
import { ExtensionViews, BroadcasterConfig, LiveConfig, Configurations, ProductManagement } from '../constants/nav-items';
import { UserDropdown } from '../user-dropdown';
import { LoginButton } from '../login-button';
import { UserSession } from '../core/models/user-session';
import { ExtensionManifest } from '../core/models/manifest';
import './component.sass';
import { MockApiDropdown } from './mock-api-dropdown';
import * as gear from '../img/gear.svg';

export interface PublicProps {
  openProductManagementHandler: Function,
  openConfigurationsHandler: Function,
  viewerHandler: Function,
  configHandler: Function,
  liveConfigHandler: Function,
  selectedView: string,
  error: string,
}

export interface ReduxStateProps {
  session?: UserSession,
  manifest?: ExtensionManifest,
  mockApiEnabled?: boolean,
}

type Props = PublicProps & ReduxStateProps;

export class RigNavComponent extends React.Component<Props> {
  private openProductManagementHandler = (): void => {
    const { session, manifest, openProductManagementHandler } = this.props;
    if ((session && session.login) && (manifest && manifest.bits_enabled)) {
      openProductManagementHandler();
    }
  }

  public render() {
    const { session, manifest, selectedView } = this.props;
    const extensionViewsClass = classNames({
      'offset': true,
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === ExtensionViews,
    });

    const broadcasterConfigClass = classNames({
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === BroadcasterConfig,
    });

    const liveConfigClass = classNames({
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === LiveConfig,
    });

    const productManagementClass = classNames({
      'top-nav-item': true,
      'top-nav-item__selected': selectedView === ProductManagement,
      'top-nav-item__disabled': !(session && session.login) || !(manifest && manifest.bits_enabled),
    });

    if (this.props.error !== '') {
      return (
        <div className='top-nav-error'>
          <a> {this.props.error} </a>
        </div>
      );
    } else {
      return (
        <div className='top-nav'>
          <div className='personal-bar'>
            <MockApiDropdown />
            <div className='personal-bar__configurations' onClick={(event) => this.props.openConfigurationsHandler()} title='Configurations'>
              <img src={gear} width={24} height={24} alt='Configurations' />
            </div>
            {manifest && <div className='personal-bar__ext-name'>
              <span>{manifest.name}</span>
            </div>}
            <div className='top-nav-item__login'>
              {(session && session.login) ? <UserDropdown session={session} /> : <LoginButton/>}
            </div>
          </div>
          <div className='top-nav__item-container'>
            <a className={extensionViewsClass} onClick={(event) => this.props.viewerHandler()}>
              Extension Views
            </a>
            <a className={broadcasterConfigClass} onClick={(event) => this.props.configHandler()}>
              Broadcaster Config
            </a>
            <a className={liveConfigClass} onClick={(event) => this.props.liveConfigHandler()}>
              Live Config
            </a>
            <a className={productManagementClass} onClick={(event) => this.openProductManagementHandler()}>
              Manage Bits Products
            </a>
          </div>
        </div>
      );
    }
  }
}
