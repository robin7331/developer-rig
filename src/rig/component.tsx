import * as React from 'react';
import './component.sass';
import { RigNav } from '../rig-nav';
import { ExtensionViewContainer } from '../extension-view-container';
import { Console } from '../console';
import { ExtensionViewDialog } from '../extension-view-dialog';
import { RigConfigurationsDialog } from '../rig-configurations-dialog';
import { EditViewDialog, EditViewProps } from '../edit-view-dialog';
import { ProductManagementViewContainer } from '../product-management-container';
import { createExtensionObject } from '../util/extension';
import { createSignedToken } from '../util/token';
import { fetchManifest, fetchExtensionManifest, fetchUserInfo } from '../util/api';
import { ExtensionViews, BroadcasterConfig, LiveConfig, Configurations, ProductManagement } from '../constants/nav-items'
import { ViewerTypes } from '../constants/viewer-types';
import { OverlaySizes } from '../constants/overlay-sizes';
import { IdentityOptions } from '../constants/identity-options';
import { MobileSizes } from '../constants/mobile';
import { RigRole } from '../constants/rig';
import { RigExtensionView, RigExtension } from '../core/models/rig';
import { Extension } from '../core/models/extension';
import { ExtensionManifest } from '../core/models/manifest';
import { UserSession } from '../core/models/user-session';
import { SignInDialog } from '../sign-in-dialog';
import { ExtensionMode, ExtensionViewType } from '../constants/extension-coordinator';

export interface ReduxStateProps {
  session: UserSession;
}

export interface ReduxDispatchProps {
  saveManifest: (manifest: ExtensionManifest) => void;
  userLogin: (userSession: UserSession) => void;
}

export interface RigProps { }

interface State {
  apiHost: string;
  clientId: string;
  secret: string;
  version: string;
  channelId: string;
  userName: string;
  mode: string;
  extensionViews: RigExtensionView[],
  manifest: ExtensionManifest;
  showSignInDialog: boolean;
  showExtensionsView: boolean;
  showConfigurations: boolean;
  showEditView: boolean;
  showProductManagementView: boolean;
  idToEdit: string;
  selectedView: string;
  extension: RigExtension;
  userId: string;
  error: string;
}

type Props = RigProps & ReduxDispatchProps & ReduxStateProps;

export class RigComponent extends React.Component<Props, State> {
  public refs: {
    extensionViewDialog: ExtensionViewDialog;
  }

  public state: State = {
    apiHost: process.env.API_HOST || 'api.twitch.tv',
    clientId: process.env.EXT_CLIENT_ID,
    secret: process.env.EXT_SECRET,
    version: process.env.EXT_VERSION,
    channelId: process.env.EXT_CHANNEL_ID,
    userName: process.env.EXT_USER_NAME,
    mode: ExtensionMode.Viewer,
    extensionViews: [],
    manifest: {} as ExtensionManifest,
    showSignInDialog: true,
    showExtensionsView: false,
    showConfigurations: false,
    showEditView: false,
    showProductManagementView: false,
    idToEdit: '0',
    selectedView: ExtensionViews,
    extension: {} as RigExtension,
    userId: '',
    error: '',
  }

  constructor(props: Props) {
    super(props);
    this.setLogin();
  }

  public componentDidMount() {
    this.initLocalStorage();
    this.fetchInitialConfiguration();
  }

  public openConfigurationsHandler = () => {
    this.setState({
      showConfigurations: true,
    });
  }

  public closeConfigurationsHandler = () => {
    this.setState({
      showConfigurations: false,
    });
  }

  public openEditViewHandler = (id:string) => {
    this.setState({
      showEditView: true,
      idToEdit: id,
    });
  }

  public closeEditViewHandler = () => {
    this.setState({
      showEditView: false,
      idToEdit: '0',
    });
  }

  public viewerHandler = () => {
    this.setState({
      mode: ExtensionMode.Viewer,
      selectedView: ExtensionViews,
      extension: {} as RigExtension,
    });
  }

  public configHandler = () => {
    this.setState({
      mode: ExtensionMode.Config,
      selectedView: BroadcasterConfig,
      extension: createExtensionObject(
        this.state.manifest,
        '0',
        ViewerTypes.Broadcaster,
        true,
        this.state.userName,
        this.state.channelId,
        this.state.secret,
        ''),
    });
  }

  public liveConfigHandler = () => {
    this.setState({
      mode: ExtensionMode.Dashboard,
      selectedView: LiveConfig,

      extension: createExtensionObject(
        this.state.manifest,
        '0',
        ViewerTypes.Broadcaster,
        true,
        this.state.userName,
        this.state.channelId,
        this.state.secret,
        ''),
    });
  }

  private openProductManagementHandler = () => {
    this.setState({
      selectedView: ProductManagement,
    });
  }

  public openExtensionViewHandler = () => {
    if (this.state.error === '') {
      this.setState({
        showExtensionsView: true,
      });
    }
  }

  public closeExtensionViewDialog = () => {
    this.setState({
      showExtensionsView: false
    });
  }

  public closeSignInDialog = () => {
    this.setState({
      showSignInDialog: false
    });
  }

  private refreshConfigurationsHandler = () => {
    const token = createSignedToken(RigRole, '', this.state.userId, this.state.channelId, this.state.secret);
    fetchExtensionManifest(this.state.apiHost, this.state.clientId, this.state.version, token)
      .then(this.onConfigurationSuccess)
      .catch(this.onConfigurationError);
  }

  public onConfigurationSuccess = (data: any) => {
    this.props.saveManifest(data.manifest);
    this.setState(data);
  }

  public onConfigurationError = (errMsg: string) => {
    this.setState({
      error: errMsg,
    });
  }

  public getFrameSizeFromDialog(dialogRef: any) {
    if (dialogRef.state.frameSize === 'Custom') {
      return {
        width: dialogRef.state.width,
        height: dialogRef.state.height
      };
    }
    if (dialogRef.state.extensionViewType === ExtensionViewType.Mobile) {
      return MobileSizes[dialogRef.state.frameSize];
    }

    return OverlaySizes[dialogRef.state.frameSize];
  }

  public createExtensionView = () => {
    const extensionViews = this.getExtensionViews();
    const linked = this.refs.extensionViewDialog.state.identityOption === IdentityOptions.Linked;
    const nextExtensionViewId = extensionViews.reduce((a: number, b: RigExtensionView) => Math.max(a, parseInt(b.id, 10)), 0) + 1;
    extensionViews.push({
      id: nextExtensionViewId.toString(),
      type: this.refs.extensionViewDialog.state.extensionViewType,
      extension: createExtensionObject(
        this.state.manifest,
        nextExtensionViewId.toString(),
        this.refs.extensionViewDialog.state.viewerType,
        linked,
        this.state.userName,
        this.state.channelId,
        this.state.secret,
        this.refs.extensionViewDialog.state.opaqueId,
      ),
      linked: linked,
      role: this.refs.extensionViewDialog.state.viewerType,
      x: this.refs.extensionViewDialog.state.x,
      y: this.refs.extensionViewDialog.state.y,
      orientation: this.refs.extensionViewDialog.state.orientation,
      frameSize: this.getFrameSizeFromDialog(this.refs.extensionViewDialog),
    });
    this.pushExtensionViews(extensionViews);
    this.closeExtensionViewDialog();
  }

  public deleteExtensionView = (id:string) => {
    this.pushExtensionViews(this.state.extensionViews.filter(element => element.id !== id));
  }

  public editViewHandler = (newViewState: EditViewProps) => {
    const views = this.getExtensionViews();
    views.forEach((element: RigExtensionView)=> {
      if (element.id === this.state.idToEdit) {
        element.x = newViewState.x;
        element.y = newViewState.y;
        element.orientation = newViewState.orientation;
      }
    });
    this.pushExtensionViews(views);
    this.closeEditViewHandler();
  }

  public render() {
    let view = (
      <div>
        <ExtensionViewContainer
          mode={this.state.mode}
          extensionViews={this.state.extensionViews}
          deleteExtensionViewHandler={this.deleteExtensionView}
          openExtensionViewHandler={this.openExtensionViewHandler}
          openEditViewHandler={this.openEditViewHandler}
          extension={this.state.extension} />
        {this.state.showExtensionsView &&
          <ExtensionViewDialog
            ref="extensionViewDialog"
            extensionViews={this.state.manifest.views}
            show={this.state.showExtensionsView}
            closeHandler={this.closeExtensionViewDialog}
            saveHandler={this.createExtensionView} />}
        {this.state.showEditView &&
          <EditViewDialog
            idToEdit={this.state.idToEdit}
            show={this.state.showEditView}
            views={this.getExtensionViews()}
            closeHandler={this.closeEditViewHandler}
            saveViewHandler={this.editViewHandler}
          />}
        {this.state.showConfigurations && <RigConfigurationsDialog
          config={this.state.manifest}
          closeConfigurationsHandler={this.closeConfigurationsHandler}
          refreshConfigurationsHandler={this.refreshConfigurationsHandler} />}
        {!this.props.session &&
          <SignInDialog
            show={this.state.showSignInDialog}
            closeSignInDialog={this.closeSignInDialog} />
        }
        <Console />
      </div>
    );

    if (this.state.selectedView === ProductManagement) {
      view = (
        <ProductManagementViewContainer clientId={this.state.clientId} />
      );
    }

    return (
      <div className="rig-container">
        <RigNav
          selectedView={this.state.selectedView}
          viewerHandler={this.viewerHandler}
          configHandler={this.configHandler}
          liveConfigHandler={this.liveConfigHandler}
          openConfigurationsHandler={this.openConfigurationsHandler}
          openProductManagementHandler={this.openProductManagementHandler}
          error={this.state.error}/>
        {view}
      </div>
    );
  }

  public getExtensionViews() {
    const extensionViewsValue = localStorage.getItem("extensionViews");
    return extensionViewsValue ? JSON.parse(extensionViewsValue) : extensionViewsValue;
  }

  public pushExtensionViews(newViews: RigExtensionView[]) {
    localStorage.setItem("extensionViews", JSON.stringify(newViews));
    this.setState({
      extensionViews: newViews,
    });
  }

  private fetchInitialConfiguration() {
    if (this.state.userName) {
      fetchManifest(
        this.state.apiHost,
        this.state.clientId,
        this.state.userName,
        this.state.version,
        this.state.channelId,
        this.state.secret
      )
      .then(this.onConfigurationSuccess)
      .catch(this.onConfigurationError);
    }
  }

  private initLocalStorage() {
    const extensionViewsValue = localStorage.getItem("extensionViews");
    if (extensionViewsValue) {
      const extensionViews = JSON.parse(extensionViewsValue);
      extensionViews.forEach((view: RigExtensionView, index: number) => view.id = (index + 1).toString());
      this.setState({ extensionViews });
    } else {
      localStorage.setItem("extensionViews", JSON.stringify([]));
    }
  }

  private setLogin() {
    const windowHash = window.location.hash;
    const rigLogin = localStorage.getItem('rigLogin');
    if (windowHash.includes('access_token')) {
      const accessTokenKey = 'access_token=';
      const accessTokenIndex = windowHash.indexOf(accessTokenKey);
      const ampersandIndex = windowHash.indexOf('&');
      const accessToken = windowHash.substring(accessTokenIndex + accessTokenKey.length, ampersandIndex);
      fetchUserInfo(accessToken)
        .then(resp => {
          const userSess = {
            login: resp.login,
            authToken: accessToken,
            profileImageUrl: resp.profile_image_url,
          }
          this.state.userName = resp.login;
          this.props.userLogin(userSess);
          localStorage.setItem('rigLogin', JSON.stringify(userSess));
          window.location.assign('/');
        })
        .catch(err => {
          this.setState({
            error: err,
          });
        });
    }
    else if (rigLogin) {
      const login = JSON.parse(rigLogin);
      this.state.userName = login.login;
      this.props.userLogin({
        login: login.login,
        authToken: login.authToken,
        profileImageUrl: login.profileImageUrl,
      });
    }
  }
}
