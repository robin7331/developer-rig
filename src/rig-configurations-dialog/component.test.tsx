import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { RigConfigurationsDialog } from './component';

describe('<RigConfigurationsDialog />', () => {
  const setupShallow = setupShallowTest(RigConfigurationsDialog, () => ({
    config: {},
    closeConfigurationsHandler: jest.fn(),
    refreshConfigurationsHandler: jest.fn()
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('expect no config', () => {
    const { wrapper } = setupShallow();
    expect(wrapper.find('.rig-configurations-view__content').text().trim()).toBe('{}');
  });

  it('should have some configurations', () => {
    const testConfig = {
      test: 'config'
    }
    const { wrapper } = setupShallow({
      config: testConfig
    });
    expect(JSON.parse(wrapper.find('.rig-configurations-view__content').text().trim())).toEqual(testConfig);
  });

  it('fires closeConfigurationsHandler when top exit button is clicked', () => {
    const { wrapper } = setupShallow();
    wrapper.find('.top-bar-container__escape').simulate('click');
    expect(wrapper.instance().props.closeConfigurationsHandler).toHaveBeenCalled();
  });

  it('fires closeConfigurationsHandler when cancel button is clicked', () => {
    const { wrapper } = setupShallow();
    wrapper.find('.bottom-bar__cancel').simulate('click');
    expect(wrapper.instance().props.closeConfigurationsHandler).toHaveBeenCalled();
  });

  it('fires refreshConfigurationsHandler when refresh button is clicked', () => {
    const { wrapper } = setupShallow();
    wrapper.find('.bottom-bar__refresh').simulate('click');
    expect(wrapper.instance().props.refreshConfigurationsHandler).toHaveBeenCalled();
  });
});
