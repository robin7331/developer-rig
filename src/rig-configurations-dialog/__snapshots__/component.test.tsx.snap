// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigConfigurationsDialog /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigConfigurationsDialog
    closeConfigurationsHandler={[Function]}
    config={Object {}}
    refreshConfigurationsHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="rig-configurations-view__background"
/>,
        <div
          className="rig-configurations-view__dialog"
>
          <div
                    className="dialog__top-bar-container"
          >
                    <div
                              className="top-bar-container__title"
                    >
                               Dev Rig Configurations 
                    </div>
                    <div
                              className="top-bar-container__escape"
                              onClick={[Function]}
                    >
                              <img
                                        alt="Close"
                                        src="close_icon.png"
                              />
                    </div>
          </div>
          <hr
                    className="dialog__divider"
          />
          <div
                    className="rig-configurations-view__content"
          >
                     
                    {}
                     
          </div>
          <hr
                    className="dialog__divider"
          />
          <div
                    className="dialog_bottom-bar"
          >
                    <div
                              className="bottom-bar__refresh"
                              onClick={[Function]}
                    >
                               Refresh 
                    </div>
                    <div
                              className="bottom-bar__cancel"
                              onClick={[Function]}
                    >
                               Cancel 
                    </div>
          </div>
</div>,
      ],
      "className": "rig-configurations-view",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "rig-configurations-view__background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="dialog__top-bar-container"
>
              <div
                            className="top-bar-container__title"
              >
                             Dev Rig Configurations 
              </div>
              <div
                            className="top-bar-container__escape"
                            onClick={[Function]}
              >
                            <img
                                          alt="Close"
                                          src="close_icon.png"
                            />
              </div>
</div>,
            <hr
              className="dialog__divider"
/>,
            <div
              className="rig-configurations-view__content"
>
               
              {}
               
</div>,
            <hr
              className="dialog__divider"
/>,
            <div
              className="dialog_bottom-bar"
>
              <div
                            className="bottom-bar__refresh"
                            onClick={[Function]}
              >
                             Refresh 
              </div>
              <div
                            className="bottom-bar__cancel"
                            onClick={[Function]}
              >
                             Cancel 
              </div>
</div>,
          ],
          "className": "rig-configurations-view__dialog",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="top-bar-container__title"
>
                   Dev Rig Configurations 
</div>,
                <div
                  className="top-bar-container__escape"
                  onClick={[Function]}
>
                  <img
                                    alt="Close"
                                    src="close_icon.png"
                  />
</div>,
              ],
              "className": "dialog__top-bar-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Dev Rig Configurations ",
                  "className": "top-bar-container__title",
                },
                "ref": null,
                "rendered": " Dev Rig Configurations ",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Close"
                    src="close_icon.png"
/>,
                  "className": "top-bar-container__escape",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Close",
                    "src": "close_icon.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dialog__divider",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "{}",
                " ",
              ],
              "className": "rig-configurations-view__content",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "{}",
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dialog__divider",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="bottom-bar__refresh"
                  onClick={[Function]}
>
                   Refresh 
</div>,
                <div
                  className="bottom-bar__cancel"
                  onClick={[Function]}
>
                   Cancel 
</div>,
              ],
              "className": "dialog_bottom-bar",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Refresh ",
                  "className": "bottom-bar__refresh",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": " Refresh ",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Cancel ",
                  "className": "bottom-bar__cancel",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": " Cancel ",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="rig-configurations-view__background"
/>,
          <div
            className="rig-configurations-view__dialog"
>
            <div
                        className="dialog__top-bar-container"
            >
                        <div
                                    className="top-bar-container__title"
                        >
                                     Dev Rig Configurations 
                        </div>
                        <div
                                    className="top-bar-container__escape"
                                    onClick={[Function]}
                        >
                                    <img
                                                alt="Close"
                                                src="close_icon.png"
                                    />
                        </div>
            </div>
            <hr
                        className="dialog__divider"
            />
            <div
                        className="rig-configurations-view__content"
            >
                         
                        {}
                         
            </div>
            <hr
                        className="dialog__divider"
            />
            <div
                        className="dialog_bottom-bar"
            >
                        <div
                                    className="bottom-bar__refresh"
                                    onClick={[Function]}
                        >
                                     Refresh 
                        </div>
                        <div
                                    className="bottom-bar__cancel"
                                    onClick={[Function]}
                        >
                                     Cancel 
                        </div>
            </div>
</div>,
        ],
        "className": "rig-configurations-view",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rig-configurations-view__background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="dialog__top-bar-container"
>
                <div
                                className="top-bar-container__title"
                >
                                 Dev Rig Configurations 
                </div>
                <div
                                className="top-bar-container__escape"
                                onClick={[Function]}
                >
                                <img
                                                alt="Close"
                                                src="close_icon.png"
                                />
                </div>
</div>,
              <hr
                className="dialog__divider"
/>,
              <div
                className="rig-configurations-view__content"
>
                 
                {}
                 
</div>,
              <hr
                className="dialog__divider"
/>,
              <div
                className="dialog_bottom-bar"
>
                <div
                                className="bottom-bar__refresh"
                                onClick={[Function]}
                >
                                 Refresh 
                </div>
                <div
                                className="bottom-bar__cancel"
                                onClick={[Function]}
                >
                                 Cancel 
                </div>
</div>,
            ],
            "className": "rig-configurations-view__dialog",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="top-bar-container__title"
>
                     Dev Rig Configurations 
</div>,
                  <div
                    className="top-bar-container__escape"
                    onClick={[Function]}
>
                    <img
                                        alt="Close"
                                        src="close_icon.png"
                    />
</div>,
                ],
                "className": "dialog__top-bar-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Dev Rig Configurations ",
                    "className": "top-bar-container__title",
                  },
                  "ref": null,
                  "rendered": " Dev Rig Configurations ",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="Close"
                      src="close_icon.png"
/>,
                    "className": "top-bar-container__escape",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Close",
                      "src": "close_icon.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dialog__divider",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "{}",
                  " ",
                ],
                "className": "rig-configurations-view__content",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "{}",
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dialog__divider",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="bottom-bar__refresh"
                    onClick={[Function]}
>
                     Refresh 
</div>,
                  <div
                    className="bottom-bar__cancel"
                    onClick={[Function]}
>
                     Cancel 
</div>,
                ],
                "className": "dialog_bottom-bar",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Refresh ",
                    "className": "bottom-bar__refresh",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": " Refresh ",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Cancel ",
                    "className": "bottom-bar__cancel",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": " Cancel ",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
